algoliasearch/client.go

```diff
@@ -258,6 +259,48 @@ func (c *client) BatchWithRequestOptions(operations []BatchOperationIndexed, opt
 	return
 }
 
+func (c *client) WaitTask(indexName string, taskID int) error {
+	return c.WaitTaskWithRequestOptions(indexName, taskID, nil)
+}
+
+func (c *client) WaitTaskWithRequestOptions(indexName string, taskID int, opts *RequestOptions) error {
+	var res TaskStatusRes
+	var err error
+
+	var maxDuration = time.Second
+	var sleepDuration time.Duration
+
+	for {
+		if res, err = c.GetStatusWithRequestOptions(indexName,
+			taskID, opts); err != nil {
+			return err
+		}
+
+		if res.Status == "published" {
+			return nil
+		}
+
+		sleepDuration = randDuration(maxDuration)
+		time.Sleep(sleepDuration)
+
+		// Increase the upper boundary used to generate the sleep
+		// duration
+		if maxDuration < 10*time.Minute {
+			maxDuration *= 2
```

Comment:
As we probably want to be bounded by 10min, why do you think of capping the value of `maxDuration` to 10min just after this line? Otherwise, if `maxDuration` was at 6min for instance, you would have to wait for 12min afterwards.